components/org.wso2.carbon.identity.login.resolver/org.wso2.carbon.identity.login.resolver.service/src/main/java/org/wso2/carbon/identity/login/resolver/service/util/LoginResolverServiceUtil.java

```diff
@@ -0,0 +1,148 @@
+import org.wso2.carbon.identity.login.resolver.service.internal.LoginResolverServiceDataHolder;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * The utility class that is used within the login resolver service module.
+ */
+public class LoginResolverServiceUtil {
+
+    private static final Log log = LogFactory.getLog(LoginResolverServiceUtil.class);
+
+    /**
+     * Private constructor to restrict from creating multiple instances of the LoginResolverServiceUtil class.
+     */
+    private LoginResolverServiceUtil() {
+
+    }
+
+    /**
+     * Retrieves the connector configuration based on the provided key and the tenant domain.
+     *
+     * @param key          The key which is used to retrieve the connector configuration for.
+     * @param tenantDomain The tenant domain which is used to retrieve the connector configuration for.
+     * @return The connector configuration that corresponds to the key and the tenant domain. An empty string if there
+     * is no matching connector configuration.
+     * @throws IdentityEventException If there is an error while retrieving the connector configuration.
+     */
+    public static String getConnectorConfig(String key, String tenantDomain) throws IdentityEventException {
+
+        try {
+            Property[] connectorConfigs;
+            IdentityGovernanceService identityGovernanceService = LoginResolverServiceDataHolder.getInstance()
+                    .getIdentityGovernanceService();
+            if (identityGovernanceService != null) {
+                connectorConfigs = identityGovernanceService.getConfiguration(new String[]{key}, tenantDomain);
+                if (connectorConfigs != null && connectorConfigs.length > 0) {
+                    return connectorConfigs[0].getValue();
+                }
+            }
+            return StringUtils.EMPTY;
+        } catch (IdentityGovernanceException e) {
+            throw new IdentityEventException(String.format("An error occurred while getting the connector " +
+                    "configurations for the property : %s", key), e);
+        }
+    }
+
+    /**
+     * Retrieves the list of claim URIs which are enabled for the login resolver on given tenant domain.
+     *
+     * @param tenantDomain The tenant domain of the user.
+     * @return The claim list for which the login resolver is enabled.
+     */
+    public static List<String> getAllowedClaimsForTenant(String tenantDomain) {
+
+        List<String> allowedClaimsList = new ArrayList<>();
+        if (StringUtils.isNotBlank(tenantDomain)) {
+            try {
+                String claimList = LoginResolverServiceUtil.
+                        getConnectorConfig(LoginResolverServiceConstants.ALLOWED_LOGIN_ATTRIBUTES, tenantDomain);
+                if (StringUtils.isNotBlank(claimList)) {
+                    claimList = StringUtils.deleteWhitespace(claimList);
```

Comment:
what will happen if ",,someclaims1, someclaim2,,," is set from the UI

Reply:
I have not specifically tested this scenario but according to the source code, I'm guessing that it will split them as follows.
```
""
""
"someclaims1"
"someclaims2"
""
""
""
```
Nevertheless, I don't think it will affect the flow except the fact that there will be a slight performance decrease.

Reply:
If the return ob of this method is used to call below method,
```
public AuthenticationResult authenticateWithIdentifier(String loginIdentifier, List<String> allowedAttributes,
                                                           Object credential, String tenantDomain)
```
Is the allowedAttributes this claim list. 
If so in the claim manager implementation, we have to add logic not calling db on empty strings. But in our current impl it's like below.
https://github.com/wso2/carbon-kernel/blob/4.9.x/core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/claim/DefaultClaimManager.java#L145

Reply:
I have avoided this call from the current source code. Fixed with https://github.com/wso2-extensions/identity-governance/pull/652/commits/8850e6ed66ee47b980c57d32cb7918c9dde6d577.